src/main/java/com/opentable/service/ServiceInfo.java

```diff
@@ -14,11 +14,151 @@
 
 package com.opentable.service;
 
+import java.net.URI;
+import java.util.Objects;
+
 /**
  * The {@link ServiceInfo} interface is intended for you to declare as a Spring Bean in your application.
  * It can be consumed by our components in order to create things like metric namespaces given information about your
  * service.  You are expected to implement your own implementations of this interface when defining your Bean.
+ * The easiest way to do so is use the Builder.
+ *
+ * For more details on these fields see https://wiki.otcorp.opentable.com/display/CP/The+service-details+Endpoint
  */
 public interface ServiceInfo {
+    /**
+     * Service Name
+     */
     String getName();
+
+    /**
+     * A description of the service
+     */
+    String description();
+
+    /**
+     * The name of the team responsible for the operation of the service.
+     */
+    String teamName();
+
+    /**
+     * The email of the team
+     */
+    String teamEmail();
+
+    /**
+     * PagerDuty contact info
+     */
+    String pagerDutyContact();
+
+    /**
+     * The github repo wherein the service code resides
+     */
+    URI repository();
+
+    /**
+     * Link to documentation, typically a wiki or README.MD
+     */
+    URI documentation();
+
+    /**
+     * Link to the runbook, typically in the wiki
+     */
+    URI runBook();
```

Comment:
I am still not really sure this is the right way to convey this information.  Putting this in code has a number of downsides, including:
* Requirement to split QA vs Prod dashboard / docs links
* Prod deployment to update contact information or Kibana links
* If the service is down, I can't look up the runbook (!)

These read more like Pantry metadata fields to me.

Reply:
That's all true, and that's an issue in general with the ServiceDetails endpoint.

Reply:
Most of the purpose of this PR is to discuss the sanity of that endpoint and using ServiceInfo as a bridge for the core data.

Reply:
> Requirement to split QA vs Prod dashboard / docs links

Yeah, this one is a bit suspect.  Imagining implementing this, I would have to inject the `AppInfo` in order to figure out whether we're running in QA or not so I can give back the right link.

> These read more like Pantry metadata fields to me.

Wasn't part of the design here that Pantry would crawl our services with this metadata and update Pantry?

Generally, of course, I'm agreed.  I think a better architecture here would be to specify a uniform naming convention for run books, etc., and then have it keyed off of a minimal set of data (e.g., service name, component ID).  Then, we don't need any metadata store for links to things like this.  We can just generate them given a key.

Reply:
In fact, if things are required to be uniformized, then we can just crawl everything listed in discovery and _generate_ all of this metadata.